tests/contest/contest/src/tests/linux_ns_path/ns_path_test.rs

```diff
@@ -0,0 +1,275 @@
+use std::fs;
+use std::os::unix::fs::MetadataExt;
+use std::os::unix::process::CommandExt;
+use std::path::PathBuf;
+use std::process::{Child, Command};
+use std::time::Instant;
+
+use anyhow::{Error, anyhow};
+use nix::sys::signal::{Signal, kill};
+use nix::unistd::Pid;
+use oci_spec::runtime::{
+    LinuxBuilder, LinuxNamespaceBuilder, LinuxNamespaceType, Spec, SpecBuilder,
+};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::utils::test_outside_container;
+
+struct WithCleanup {
+    child: Child,
+}
+
+impl WithCleanup {
+    fn new(child: Child) -> Self {
+        Self { child }
+    }
+}
+
+impl Drop for WithCleanup {
+    fn drop(&mut self) {
+        match kill(Pid::from_raw(-(self.child.id() as i32)), Signal::SIGKILL) {
+            Ok(()) => {}
+            Err(_e) => {} // Not sure what else can be done here
+        }
```

Comment:
Should we wait for the child processes is killed to avoid making zombi processes?